### ISSUE TITLE
Quantity .positive? and negative? failed

### ISSUE THREAD

**Issue by jimtng** [boc-tothefuture/openhab-jruby#255]
Using #positive? and #negative? on quantity, and NumberItem with quantity failed with an error. I would expect that we should be able to use .positive? and negative? on a quantity / numberitem.

Jruby library 3.6.3 on openhab 3.1M4

```
  Scenario Outline: Quantity can use positive? and negative?
    Given code in a rules file
      """
        logger.info("Result is #{<quantity>}")
      """
    When I deploy the rules file
    Then It should log 'Result is <result>' within 5 seconds
    Examples:
      | quantity                          | result |
      | Quantity.new('50 °F').positive?   | true   |
      | Quantity.new('-100 °F').positive? | false  |
      | Quantity.new('50 °F').negative?   | false  |
      | Quantity.new('-100 W').negative?  | true   |
```

Output:
```
08:26:04.232 [INFO ] [jsr223.jruby.Module                  ] - OpenHAB JRuby Scripting Library Version 3.6.3 Loaded
08:26:04.233 [INFO ] [jsr223.jruby.Object                  ] - Processing Started for 5107bace-d831-40ed-906d-83efe6ec4c04
08:26:04.235 [TRACE] [3.jruby.OpenHAB::DSL::Types::Quantity] - Comparing -100 W to 0
08:26:04.235 [TRACE] [3.jruby.OpenHAB::DSL::Types::Quantity] - Coercing -100 W as a request from Integer
08:26:04.236 [TRACE] [3.jruby.OpenHAB::DSL::Types::Quantity] - Unitizing (-100 W) and (0)
08:26:04.250 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab-jruby/tmp/openhab/conf/automation/jsr223/ruby/personal/cucumber_test20210604-21478-q14uhr.rb': org.jruby.embed.EvalFailedException: java.lang.IllegalArgumentException: Can not compare incompatible units.
```

**Comment by pacive** [comment]
This points to a problem I had when working on migrating the UoM library: 
The `unitize` method only converts units if the `operation` parameter is `*` or `/`, so it doesn't actually do anything when called from the `<=>` method. Also, when it does convert, the `convert_unit` method checks the  `unit?` method, which only returns truthy if the operation is called in a `unit() {}` block, otherwise it does nothing. Is this really how it's supposed to work?

I had to struggle to get my head around all the twists and turns here, and couldn't really grasp the rationale for all of it, so perhaps I created some further issues when trying to get it to work.

Did this work in previous versions (< 3.6.1) @jimtng?
